addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -687,6 +704,12 @@ def _import_fattura_pa(self, tree, invoice):
 
             new_invoice = invoice_form
 
+            # Information related to the purchase order <2.1.2>
+            elements = body_tree.xpath('//DatiGenerali/DatiOrdineAcquisto')
+            if elements:
+                po_refs = [po_ref_element.text for element in elements for po_ref_element in element.xpath(".//IdDocumento")]
+                new_invoice._find_and_set_purchase_orders(po_refs, partner.id, new_invoice.amount_total, timeout=4)
```

Comment:
I don't think you can call it here, because it will already be called if you put it manually in the invoice here: https://github.com/odoo/odoo/blob/ebc8b007ecb375d20c566b6e6ecc3f6749ebaa2a/addons/account_edi/models/account_journal.py#L91

Why not put it in the invoice_origin as well?  But I am afraid we will have a problem with the amount_total...

Reply:
Ok for the `invoice_origin`, didn't know about it.

The code you've shown doesn't really belong to `account_edi.account_journal`...
It doesn't also check if the link has already been made.
It should either be in
    `account.account_move.create_from_attachment` 
    or `account.ir_attachment.create_move`
so that it can also be called when we upload from the journal without using the EDI.

Action code should be in the wizard `account_tour_upload_bill` wizard
https://github.com/odoo/odoo/blob/ebc8b007ecb375d20c566b6e6ecc3f6749ebaa2a/addons/account/wizard

![image](https://user-images.githubusercontent.com/1665365/221536897-f8bdb42a-3374-4cb4-a5f8-1ab64489df39.png)

Reply:
I've put a condition on the generic method to check that the link's not already done. We wait for a bit of refactoring, as the `account_edi.account_journal` position may not be the best one.